app/Utils/Traits/GeneratesCounter.php

```diff
@@ -663,8 +671,16 @@ private function resetCompanyCounters($company)
         $settings->recurring_expense_number_counter = 1;
         $settings->purchase_order_number_counter = 1;
 
+        if( config('ninja.reset_all_counter') ) {
+            $settings->vendor_number_counter = 1;
+        }
+
         $company->settings = $settings;
         $company->save();
+
+        if ($new_reset_date->lte(now())) {
+            return $this->resetCompanyCounters($company);
+        }
```

Comment:
Can you explain the reasoning behind this? There is an early return at the beginning of the method that would catch this, so I am unsure why we would reloop this?

For PRs of this nature that affect the core logic of the app, we would need to have additional tests present also that confirm functionality.

Reply:
Yeah sure :-) Setting reset counter to `daily` base doesn't mean you're resetting the counter every day (e.g. via cron or something like that). Currently, you only trigger the counter reset while creating a new number. So skipping some days creating new entries (invoices, offer, ...) means the reset counter date (which you can read from admin settings) is may lots in past. This recursion helps to update the reset counter date to latest valid date. This will not break anything else, because it's run as before in defined intervall (daily, weekly, monthly, yearly...), we're using `$new_reset_date` here to check recusion validity. Same above :-)